redis/asyncio/client.py

```diff
@@ -596,29 +625,87 @@ async def _disconnect_raise(self, conn: Connection, error: Exception):
         ):
             raise error
 
+    def _cache_invalidation_process(
+        self, data: List[Union[str, Optional[List[str]]]]
+    ) -> None:
+        if data[1] is not None:
+            for key in data[1]:
+                self.client_cache.invalidate(key)
+            else:
+                self.client_cache.flush()
+
+    async def _get_from_local_cache(self, command: str):
+        """
+        If the command is in the local cache, return the response
+        """
+        if (
+            self.client_cache is None
+            or command[0] in self.cache_blacklist
+            or command[0] not in self.cache_whitelist
+        ):
+            return None
+        while not self.connection._is_socket_empty():
+            await self.connection.read_response(push_request=True)
+        return self.client_cache.get(command)
+
+    def _add_to_local_cache(self, command: str, response: ResponseT, keys: List[KeysT]):
```

Comment:
command is a string... but command[0] is a check for in cache_whitelist/blacklist